bundles/org.openhab.binding.danfossairunit/src/main/java/org/openhab/binding/danfossairunit/internal/ValueCache.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.danfossairunit.internal;
+
+import java.util.Calendar;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.core.types.State;
+
+/**
+ * The {@link ValueCache} is responsible for holding the last value of the channels for a
+ * certain amount of time {@link ValueCache#durationMs} to prevent unnecessary event bus updates if the value didn't
+ * change.
+ *
+ * @author Robert Bach - Initial contribution
+ */
+@NonNullByDefault
+public class ValueCache {
```

Comment:
Rather than implementing your own cache. I would recommend using the one that already comes with the OH core.
You can find details [here](https://www.openhab.org/docs/developer/utils/tools.html)

Reply:
I think you might have missed this suggestion from my first review so I'll copy it here:

Rather than implementing your own cache. I would recommend using the one that already comes with the OH core.
You can find details [here](https://www.openhab.org/docs/developer/utils/tools.html)
The class I think would be most relevant to you would be `org.eclipse.smarthome.core.cache.ExpiringCacheMap`.